### PR TITLE
Review app structure and change less to sass

### PR DESCRIPTION
**Pull request by maxime1992**
As we talked in branch master, here are the changes.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -58,18 +58,18 @@ function lesslint() {
 }
 
 function less() {
```
Comment (ghpabs) [inline gulpfile.js]: Update the task name.

File: gulpfile.js [hunk 2]
```diff
@@ -90,14 +90,14 @@ var tsProject = plugins.typescript.createProject('tsconfig.json', {
 });
 
 function ts() {
-	var tsResult = gulp.src('src/**/*.ts')
+	var tsResult = gulp.src('src/app/**/*')
```
Comment (ghpabs) [inline gulpfile.js]: Missing `.ts` extension.

File: gulpfile.js [hunk 3]
```diff
@@ -158,8 +158,8 @@ function protractor() {
 }
 
 function watch() {
-	gulp.watch('src/**/*.ts', gulp.series(tslint, ts, 'unit'));
-	gulp.watch('src/**/*.less', gulp.series(lesslint, less));
+	gulp.watch('src/app/**/*.ts', gulp.series(tslint, ts, 'unit'));
+	gulp.watch('src/less/**/*.less', gulp.series(lesslint, less));
```
Comment (ghpabs) [inline gulpfile.js]: Update `less` folder, task `lesslint` and `less`.
Also; `lesslint` needs to be rewritten using a `sasslint` plugin.

Comment (maxime1992) [inline gulpfile.js]: I was tired, sorry about that commit it's not a clean one.
I will correct everything when i get home.

File: gulpfile.js [hunk 4]
```diff
@@ -58,18 +58,18 @@ function lesslint() {
 }
 
 function less() {
-	return gulp.src('src/**/*.less')
+	return gulp.src('src/sass/**/*.scss')
```
Comment (ghpabs) [inline gulpfile.js]: Let's set the glob to `'src/**/*.scss'` as we could potentially have SASS files under a specific component in `/app`. Same apply to other globs in the PR.

File: gulpfile.js (hunk 3 above)
Comment (ghpabs) [inline gulpfile.js]: It's okay mate. It shouldn't be hard to fix it. :-)

File: gulpfile.js (hunk 4 above)
Comment (maxime1992) [inline gulpfile.js]: For sass it makes sense, but for ts, we should only have it into /src/app/ right ?

**Comment by maxime1992** [comment]
I think it would be cool to have a gitter. 
As the main repo is yours, i think it's better if you create it instead me with my fork.
https://gitter.im/home this way we can talk without polluting PR here if we want to discuss about what's next etc ...

**Comment by ghpabs** [comment]
Done. See https://gitter.im/ghpabs/angular2-seed.

File: gulpfile.js (hunk 4 above)
Comment (ghpabs) [inline gulpfile.js]: Fair enough. We can assume all `.ts` will be written in `/src/app`. :+1:

File: package.json [hunk 5]
```diff
@@ -21,12 +21,13 @@
     "gulp-connect": "^2.2.0",
     "gulp-if": "^2.0.0",
     "gulp-inject": "^3.0.0",
-    "gulp-less": "^3.0.5",
     "gulp-load-plugins": "^1.1.0",
     "gulp-preprocess": "^2.0.0",
     "gulp-protractor": "^2.1.0",
     "gulp-recess": "^1.2.0",
```
Comment (ghpabs) [inline package.json]: Remove `gulp-recess` - unused dep.

File: src/index.html [hunk 6]
```diff
@@ -2,13 +2,25 @@
...
 	<!-- endinject -->
 
 	<!-- inject:css -->
 	<!-- endinject -->
+
+	<!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
+	<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
+	<!--[if lt IE 9]>
+	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
+	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
```
Comment (ghpabs) [inline src/index.html]: `respond.js` is for IE 6-8 but Angular 2 only supports IE 9+ so I think it's safe to say we won't need it.

File: README.md [hunk 7]
```diff
@@ -49,5 +49,11 @@ Use `--dev` and `--prod` modifiers for development and production targets where
 - `$ gulp build serve` is equivalent to `$ gulp build serve --dev`.
 - `$ gulp build serve --prod`.
 
+## Enhancements
+- Expand app and use new Angular 2 components - Eg. `HTTP` and `Router`. See [angular2-tour-of-heroes](https://github.com/johnpapa/angular2-tour-of-heroes).
+- Upgrade `angular2` dependency to latest.
+- Remove `remap-istanbul` dependency when Istanbul v1 is released as it'll have sourcemaps built-in.
+- Expand unit tests.
```
Comment (ghpabs) [inline README.md]: Enhancements section was removed in my previous commit.

**Comment by ghpabs** [comment]
We're are almost good for a merge here. See comments above.

Thanks!

File: package.json (hunk 5 above)
Comment (maxime1992) [inline package.json]: Weird, i don't have it in my package.json !

File: README.md (hunk 7 above)
Comment (maxime1992) [inline README.md]: I had a conflict, i might have mistaken !
Removing it.

File: src/index.html (hunk 6 above)
Comment (maxime1992) [inline src/index.html]: You're right ! I'm removing it.

File: package.json (hunk 5 above)
Comment (maxime1992) [inline package.json]: I didn't noticed but in red with minus in front of the line means i deleted it !
That's why i don't have it anymore.

Comment (ghpabs) [inline package.json]: Weird! Maybe you didn't commit/push it? It still shows the line here.

**Comment by ghpabs** [comment]
Fix #12
Fix #19

**Pull request merged**